Fix branch list including the upstream and any detached head.

JGRennison opened the pull request:
If the upstream is a branch instead of a remote, it would be included
in the list of branches to rebase, likewise if a branch had already
been merged. Exclude those using --no-merged.
If no branch is checked out, and the detached head is contained by
$root, git branch will add it to the list, so filter it out.

nornagon commented on git-rebase-all:
```diff
@@ -45,7 +45,7 @@ if [ $# -eq 2 ]; then
 
   # Fetch all the branches that contain the root of the subtree
-  branches=$(git branch --contains $root | cut -c 3-)
+  branches=$(git branch --contains $root --no-merged $new_upstream | grep -v '* (HEAD' | cut -c 3-)
```
Make this `* (HEAD detached` so it's clear what it's filtering out? I wonder if there's a more flexible command for listing branches... `git for-each-ref refs/heads` perhaps?

JGRennison committed:
Fix branch list including the upstream and any detached head.

If the upstream is a branch instead of a remote, it would be included
in the list of branches to rebase, likewise if a branch had already
been merged. Exclude those using --no-merged.
Use git for-each-ref instead of git branch to avoid any detached
head being added to the list.

JGRennison commented:
`git for-each-ref refs/heads` turns out to do just what we need, so I've updated the pull request to use that instead.

nornagon commented:
Nice :) Thanks!
